python_modules/dagster/dagster/_core/storage/fs_io_manager.py

```diff
@@ -62,14 +62,12 @@ def asset1():
         def asset2(asset1):
             return asset1[:5]
 
-        @repository
-        def repo():
-            return with_resources(
-                [asset1, asset2],
-                resource_defs={
-                    "io_manager": fs_io_manager.configured({"base_dir": "/my/base/path"})
-                },
-            )
+        defs = Definitions(
+            assets=[asset1, asset2],
+            resources={
+                "io_manager": fs_io_manager.configured({"base_dir": "/my/base/path"})
```

Comment:
Actually are new Pythonic variants of these avaiable? cc: @jamiedemaria 

Also remove unused imports (e.g. with_resources)

Reply:
I dont think the three io managers in this pr have been updated yet

Reply:
although working on gcs now https://github.com/dagster-io/dagster/pull/13653